scipy/stats/_stats.pyx

```diff
@@ -0,0 +1,49 @@
+cimport cython
+from numpy cimport ndarray, int64_t, intp_t
+import numpy as np
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef inline void bit_inc(intp_t[:] arr, intp_t i):
+    cdef:
+        intp_t size = arr.size
+
+    while i < size:
+        arr[i] += 1
+        i += i & -i
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef inline intp_t bit_acc(intp_t[:] arr, intp_t i):
+    cdef:
+        intp_t out = 0
+
+    while i != 0:
+        out += arr[i]
+        i -= i & -i
+
+    return out
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def _kendall_condis(ndarray[intp_t, ndim=1] x,
+                    ndarray[intp_t, ndim=1] y):
+    cdef:
+        ndarray[intp_t, ndim=1] arr = np.zeros(y.max() + 1, dtype=np.intp)
```

Comment:
```Cython
    cdef:
        intp_t[:] arr = np.zeros(y.max() + 1, dtype=np.intp)
```

This is needed to ensure ```bit_inc``` and ```bit_acc``` is called efficiently. Otherwise we get a buffer acquisition on each call when the ```ndarray``` is cast to a typed memoryview.